component/attr_cache/attr_cache.go

```diff
@@ -752,7 +708,20 @@ func (ac *AttrCache) WriteFile(options internal.WriteFileOptions) (int, error) {
 		ac.cacheLock.RLock()
 		defer ac.cacheLock.RUnlock()
 		// TODO: Could we just update the size and mod time of the file here? Or can other attributes change here?
-		ac.invalidatePath(options.Handle.Path)
+
+		var toBeInvalid *attrCacheItem
+		var err error
+		if attr == nil {
+			toBeInvalid, err = ac.cacheMap.get(options.Handle.Path)
+		} else {
+			toBeInvalid, err = ac.cacheMap.get(attr.Path) //attr is nil here for TestWriteFileDoesNotExist()
+		}
```

Comment:
```suggestion
		toBeInvalid, err := ac.cacheMap.get(attr.Path)
```
Now that you've fixed the mock calls in the test, you don't need to check for attr == nil here.
You could also remove the same check above (after the `if er != nil` block).